patches/server/0878-Fix-HumanEntity-drop-not-updating-the-client-inv.patch

```diff
@@ -0,0 +1,27 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sun, 10 Oct 2021 18:18:01 -0700
+Subject: [PATCH] Fix HumanEntity#drop not updating the client inv
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+index e7108dd70f1eb91f38e04ad71e595bc679560f75..f71424758ec95a570f127ca28654f304d4d9e7ed 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+@@ -715,8 +715,14 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
+     // Paper end
+     @Override
+     public boolean dropItem(boolean dropAll) {
+-        if (!(this.getHandle() instanceof ServerPlayer)) return false;
+-        return ((ServerPlayer) this.getHandle()).drop(dropAll);
++        // Paper start - notify client of remote slot change
++        if (!(this.getHandle() instanceof ServerPlayer player)) return false;
++        boolean success = player.drop(dropAll);
```

Comment:
What is preventing us from technically returning early here ?
`success` should be false only if no item was dropped whatsoever, so we could skip some logic here unless I am missing something.

Reply:
That's the purpose of the PR. To fix the desync. The server can cancel the drop when the client expects it to drop, so we need to resync.

Reply:
Was it ? At least in my testing the issue was that the client specifically does *not* expect an item drop and hence
does not update the inventory.

> When use the method the item drop but still show in hotbar, but if you click the item in hotbar then disapear.
The issue seems to match that.

Reply:
Yeah, that sounds right. I guess I totally forgot what this was about. The cancellation logic for PlayerDropItemEvent handles updating the inventory.